decode/clarity.ts

```diff
@@ -26,8 +26,18 @@ export function decode(input: string): DecodedPayload {
     // Sort encoded events by time to simplify summary computation
     let encoded: Token[][] = json.d.sort((a: Token[], b: Token[]) => (a[0] as number) - (b[0] as number));
 
-    if (payload.envelope.version !== version) {
-        throw new Error(`Invalid Clarity Version. Actual: ${payload.envelope.version} | Expected: ${version} | ${input.substr(0, 250)}`);
+    // Check if the incoming version is compatible with the current running code
+    // We do an exact match for major, minor and path components of the version.
+    // However, the beta portion of the version can be either same, one less or one more.
+    // This ensures we are backward and forward compatible with upto one version change.
+    let jsonVersion = parseVersion(payload.envelope.version);
+    let codeVersion = parseVersion(version);
+
+    if (jsonVersion.major !== codeVersion.major ||
+        jsonVersion.minor !== codeVersion.minor ||
+        jsonVersion.patch !== codeVersion.patch ||
+        Math.abs(jsonVersion.beta - codeVersion.beta) > 1) {
```

Comment:
You should change this to be more compartible with semantic versioning. So if the major version is the same and the minor version is the same or higher you should decode it. 
It would be good to start avoiding the b21 type numbers and just use 0.21.0 or something similar (where 0 is major, 21 is minor and 0 is patch).

Reply:
We will need to go with 1.X.Y right now. See here: https://www.npmjs.com/package/clarity-js?activeTab=versions - we already used 0.X.Y for previous clarity-js. The idea was that we will go to normal usage once we get out of beta. May be now is the time and I should snap sarveshn/vnext back to master and stop publishing beta versions? We will sync more offline.

Reply:
Synced offline. Closing this for now. We will go with 1.X.Y when time is right.